snapcraft/internal/lxd.py

```diff
@@ -141,7 +140,7 @@ def execute(self, step='snap', args=None):
             command = ['snapcraft', step]
             if step == 'snap':
                 command += ['--output', self._snap_output]
-            if self._host_arch != self._project_options.deb_arch:
+            if self._project_options.target_arch_specified:
```

Comment:
I was going to argue that it wouldn't solve the problem, but then I [made another PR](https://github.com/snapcore/snapcraft/pull/1498).